src/actions/attack.cpp

```diff
@@ -1679,7 +1681,7 @@ bool bool_leadership(const std::string& ability,const unit_map& units, const map
 	unit_ability_list abil = un->get_abilities(ability, weapon, opp_weapon);
 	for(unit_ability_list::iterator i = abil.begin(); i != abil.end();) {
 		const std::string& active_on = (*i->first)["active_on"];
-		if(!(active_on.empty() || (attacker && active_on == "offense") || (!attacker && active_on == "defense")) || !up->ability_filter_opponent(ability, *i->first, opp_loc)) {
+		if(!(active_on.empty() || (attacker && active_on == "offense") || (!attacker && active_on == "defense")) || !up->ability_filter_fighter(ability, "filter_opponent", *i->first, opp_loc) || ( attacker && !un->ability_filter_fighter(ability, "filter_attacker" , *i->first, loc)) || ( !attacker && !up->ability_filter_fighter(ability, "filter_attacker" , *i->first, opp_loc)) || (!attacker && !un->ability_filter_fighter(ability, "filter_defender", *i->first, loc)) || (attacker && !up->ability_filter_fighter(ability, "filter_defender", *i->first, opp_loc))) {
```

Comment:
I noticed this line does not call `ability_filter_matches` but the other line does not. I'm assuming there is a good reason for this?

Reply:
ability_filter_matches used for [filter_base_value] filter, it is no use in bool function.

Reply:
Ah okay, I think that makes sense.